### PR TITLE
env_config: add `HOMEBREW_BUNDLE_USER_CACHE`.

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
Used to configure `BUNDLE_USER_CACHE`.

Changes approach in https://github.com/Homebrew/brew/pull/17324 where I saw some local issues if e.g. the directory wasn't created or had weird permissions.

CC @dkav that you'll need to set this variable now.

### CODE REVIEW

**Comment by carlocab** [comment]
> CC @dkav that you'll need to set this variable now.

Won't be needed for users who set `BUNDLE_USER_CACHE`; `brew` will use that env variable to set the `HOMEBREW_` version

File: Library/Homebrew/env_config.rb [hunk 1]
```diff
@@ -91,6 +91,9 @@ module EnvConfig
         description:  "Use this as the browser when opening project homepages.",
         default_text: "`$BROWSER` or the OS's default browser.",
       },
+      HOMEBREW_BUNDLE_USER_CACHE:                {
+        description: "If set, use this directory as the `bundle`(1) user cache.",
```
Comment (carlocab) [inline Library/Homebrew/env_config.rb:95]: Should we default to a location inside `HOMEBREW_CACHE`? Otherwise we would write to `HOME`, which is weird.

Comment (MikeMcQuaid) [reply Library/Homebrew/env_config.rb:95]: I don't understand the request, asking for a `HOMEBREW_CACHE` default is just asking for this PR to be closed and keep https://github.com/Homebrew/brew/pull/17324 functionality?

I don't think we should have or need a default at all. The existing logic was fine for everyone except dkav. 

Alternatively: we rip out all this logic and make https://github.com/Homebrew/brew/issues/17276 a WONTFIX. Really not convinced this is worth the amount of effort that's gone into it and definitely not worth any more.

Comment (carlocab) [reply Library/Homebrew/env_config.rb:95]: > I don't understand the request, asking for a `HOMEBREW_CACHE` default is just asking for this PR to be closed and keep #17324 functionality?

Not necessarily; we can simply add a `default:` key here. Will make an in-line suggestion below.

> I don't think we should have or need a default at all. The existing logic was fine for everyone except dkav.

Ok, that's fine! I just agreed with you when you said not writing to `HOME` would be ideal. But I'm not hugely bothered by it, so we can leave this as it is.

Comment (carlocab) [reply Library/Homebrew/env_config.rb:95]:
```suggestion
        description: "If set, use this directory as the `bundle`(1) user cache.",
        default:     "#{HOMEBREW_DEFAULT_CACHE}/bundler_cache"
```

**Review by carlocab — APPROVED** [review]

File: Library/Homebrew/env_config.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/env_config.rb:95]: > I just agreed with you when you said not writing to `HOME` would be ideal. But I'm not hugely bothered by it, so we can leave this as it is.

Sorry, yeh, provided insufficient context here: there's additional complexity from not writing to HOME e.g. permissions, directory not existing, etc. that means that I think it's not worthwhile given the level of demand we've seen (one person 😅)

**Comment by MikeMcQuaid** [comment]
Thanks @carlocab!

**Pull request merged**